src/fromager/settings_model.py

```diff
@@ -0,0 +1,136 @@
+import typing
+
+import yaml
+from packaging.utils import NormalizedName, canonicalize_name
+from pydantic import (
+    BaseModel,
+    BeforeValidator,
+    Field,
+    HttpUrl,
+    ValidationInfo,
+    field_validator,
+)
+
+from . import overrides
+
+# build variant
+Variant = typing.NewType("Variant", str)
+
+# Package validates and transforms name to canonicalized name
+Package = typing.Annotated[
+    NormalizedName,
+    BeforeValidator(lambda pkg: canonicalize_name(pkg, validate=True)),
```

Comment:
To be backwards compatible, we need to use `overrides.pkgname_to_override_module()`.